src/Statement/StatementList.php

```diff
@@ -87,7 +87,13 @@ public function getPropertyIds() {
 	}
 
 	public function addStatement( Statement $statement ) {
-		$this->statements[] = $statement;
+		if ( $statement->getGuid() !== null
+			&& !array_key_exists( $statement->getGuid(), $this->statements )
+		) {
+			$this->statements[$statement->getGuid()] = $statement;
+		} else {
+			$this->statements[] = $statement;
```

Comment:
I understand and accept your opposition, even if I'm missing actual arguments in what you write. I'm totally fine with closing this RFC unmerged, along with #446.